components/fraud-proof-vm/fraud_proof_vm.md

```diff
@@ -0,0 +1,50 @@
+# Fraud Proof VM
+
+A fraud proof VM (FPVM) is a virtual machine which can be evaluated on-chain to settle a disputed block hash assertion. In this spec we assume the fraud proof VM is evaluated using a multi-round dispute game similar to [Truebit](https://people.cs.uchicago.edu/~teutsch/papers/truebit.pdf).
+
+## Key Properties
+
+A Fraud proof VM must satisfy the following properties to be suitable for a multi-round dispute game:
+1. Program execution can be split up into deterministic state transitions.
+2. Each state transition can be evaluated on-chain with less than or equal to a constant size gas allocation (eg. 10 million L1 gas).
+3. The FPVM must be executable off-chain in such a way that state checkpoints (ie. merklized commitments to the VM state) can be generated.
+
+## Usage
+
+### Initialization
+
+The fraud proof VM is initialized with:
+1. an executable binary; and
+2. an initialization hash for the preimage oracle.
```

Comment:
What exactly is this hash? I suspect it is the last L1 block hash against which all proofs must be proven?